torchao/utils.py

```diff
@@ -221,11 +221,12 @@ def decorator(fn):
             op_name = fn.__name__[1:]
             schema = op_name + infer_schema(fn, mutates_args={})
             lib.define(schema)
-            lib.impl(op_name, fn, "CompositeImplicitAutograd")
+            lib.impl(op_name, fn, dispatch_key)
 
             lib_namespace = lib.ns
             op = getattr(getattr(torch.ops, lib_namespace), op_name)
-            register_decomposition([op])(fn)
+            if dispatch_key == "CompositeImplicitAutograd":
+                register_decomposition([op])(fn)
```

Comment:
Feeling adding a flag in `_register_custom_op` to indicate whether this op should be decomposed will be clearer, cc @jerryzh168 what's your suggestions?